authentication/allowed_hosts_validator.go

```diff
@@ -36,14 +49,24 @@ func (v *AllowedHostsValidator) SetAllowedHosts(hosts []string) {
 	}
 }
 
+// SetAllowedHostsErrorCheck sets the list of valid hosts with error checking.
+func (v *AllowedHostsValidator) SetAllowedHostsErrorCheck(hosts []string) error {
+	v.validHosts = make(map[string]bool, len(hosts))
+	if len(hosts) > 0 {
+		for _, host := range hosts {
+			if strings.HasPrefix(host, "http") {

```

Comment:
Do you think we should match the entire protocol section as `http://` and `https://` instead? On the unlikely off chance that someone has the host starting with these characters('http.microsoft.com')?

Reply:
OK. I will fix it.